### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by nikitagromov**
[Link](https://discuss.elastic.co/t/query-string-query-doesnt-work-when-field-has-customer-analyzer/26320) to ES forum 
query_string doesn't apply custom analyzer

### CODE REVIEW

File: nefertari/elasticsearch.py [hunk 1]
```diff
@@ -759,19 +759,26 @@ def build_search_params(self, params):
...
+            if query['must']:
+                _params['body'] = {'query': {'bool': query}}
             else:
-                _params['body'] = {"query": {"match_all": {}}}
+                _params['body'] = {'query': {'match_all': {}}}
         else:
-            _params['body'] = params['body']
+            # using raw elasticsearch body requests is not secure
+            # params['body'] = params['body']
+            pass
```
Comment (nmenardg) [inline nefertari/elasticsearch.py]: So what goes in `_params['body']` then?

File: nefertari/es_query.py [hunk 2]
```diff
@@ -7,6 +7,25 @@ def pop(self, index=None):
         return self.es_keywords[super(OperationStack, self).pop()]
 
 
+def apply_analyzer(params, doc_type, get_document_cls):
+    documents = doc_type.split(',')
+    document_classes = [(get_document_cls(document), document) for document in documents]
```
Comment (nmenardg) [inline nefertari/es_query.py]: Why do a list comp here if you're iterating on the items again in a `for`?

**Review by nmenardg — APPROVED** [review]

**Review by nmenardg — APPROVED** [review]

File: nefertari/elasticsearch.py [hunk 3]
```diff
@@ -758,38 +758,45 @@ def build_search_params(self, params):
...
-
-                log.debug('Parsed ES request body {body}'.format(body=_params['body']['query']))
+            _params['body'] = {}
+
+            try:
+                _params['body']['query'] = compile_es_query(params)
+            except Exception:
+                raise JHTTPBadRequest('Bad query string for {params}'
+                                        .format(
+                                                params=_params['body']['query']['query_string']['query']))
```
Comment (nmenardg) [inline nefertari/elasticsearch.py:798]: Why not just let the exception bubble up by itself?

Comment (nikitagromov) [reply nefertari/elasticsearch.py:798]: I think there didn't informant for user to know that ValueError raised

**Review by nmenardg — APPROVED** [review]

**Review by nmenardg — APPROVED** [review]

**Pull request merged**